### ISSUE TITLE
Добавить preloader на время переключения вариаций с Двух на Четверых и обратно

### ISSUE THREAD

**Issue by IgorLesnevskiy** [Kreozot/adentro#2]
Когда переключаешь вариации с двух на четверых или обратно во время исполнения трека - некоторое время происходит подгрузка, при этом отображается белый экран. Это смущает. Было бы круто добавить какой-нибудь прелоадер под это дело.

**Comment by IgorLesnevskiy** [comment]
Если ты принимаешь пулл-реквесты, я мог бы немного поконтрибьютить по мелочи

**Comment by Kreozot** [comment]
Да, принимаю конечно. Только нужно истинную причину исправлять.
На самом деле белый экран показывается не потому, что загрузка долгая, а потому что анимация переключается только в момент начала следующего элемента. Вообще работа с анимацией пока не слишком чёткая. И переключение с одной анимации на другую - одна из известных проблем.

По-хорошему, нужно продумать механизм старта анимации с нужного куска. Пока что анимации запускаются только с начала. Этого не просто добиться из-за того, что анимация элемента - это одна функция, которая может вызывать цепочку других через промисы. Одно из решений - при анимации проверять, не вышли ли мы за пределы отведённого куска и если вышли, то резолвить промис, не запуская анимацию, тем самым запуская анимацию только оставшихся кусков.

**Comment by Kreozot** [comment]
Вот написал я это и подумал, что пока что можно и прелоадер воткнуть =] Так что с удовольствием приму такой реквест =]

**Comment by IgorLesnevskiy** [comment]
В выходные оформлю.

**Comment by GarikFF** [comment]
https://github.com/Kreozot/adentro/pull/6
